pkg/server/endpoints/config.go

```diff
@@ -107,6 +107,10 @@ func (c *Config) maybeMakeBundleEndpointServer() Server {
 	var serverAuth bundle.ServerAuth
 	if c.BundleEndpoint.ACME != nil {
 		serverAuth = bundle.ACMEAuth(c.Log.WithField(telemetry.SubsystemName, "bundle_acme"), c.Catalog.GetKeyManager(), *c.BundleEndpoint.ACME)
+	} else if c.BundleEndpoint.DiskCertManager != nil {
+		serverAuth = c.BundleEndpoint.DiskCertManager
+		// Start watching for file changes
+		go c.BundleEndpoint.DiskCertManager.WatchFileChanges(ctx)
```

Comment:
This goroutine shouldn't be launched yet. Rather, `maybeMakeBundleEndpointServer` could return a "task" that that is set on the Endpoints struct, and then that task can be managed by Endpoints.ListenAndServe (see the other tasks conditionally started in ListenAndServe).